### PR TITLE
Switch Modal to patternfly

### PR DESCRIPTION
**Pull request by gavin-stackrox**
I also simplified the Modal component user contents to remove a lot of tailwind.

Screenshot_2023-09-22_12-38-47

Screenshot_2023-09-22_12-41-39

### CODE REVIEW

File: ui/src/containers/ClusterInfoPage/DownloadArtifactsModal.tsx [hunk 1]
```diff
@@ -15,16 +15,13 @@ type ArtifactsListProps = {
...
+    <List className="pf-u-mb-md">
       {artifacts.map((artifact) => (
-        <li key={artifact.Name}>
-          <a href={artifact.URL} className="underline text-blue-500">
-            {artifact.Name}
-          </a>{' '}
-          - {artifact.Description}
-        </li>
+        <ListItem key={artifact.Name}>
+          <a href={artifact.URL}>{artifact.Name}</a> - {artifact.Description}
```
Comment (pedrottimark) [inline ui/src/containers/ClusterInfoPage/DownloadArtifactsModal.tsx:21]: From the picture, it looks like no link formatting?

What do you think are pro and con of list in which items have two parts versus 2 column table?

Comment (gavin-stackrox) [reply ui/src/containers/ClusterInfoPage/DownloadArtifactsModal.tsx:21]: > From the picture, it looks like no link formatting?

Yes. I think that is from tailwind and patternFly but my css inspection skills are lacking.

> What do you think are pro and con of list in which items have two parts versus 2 column table?

I have no preference.

File: ui/src/components/Modal.tsx [hunk 2]
```diff
@@ -1,13 +1,12 @@
 import React, { ReactElement, ReactNode } from 'react';
-import ReactModal from 'react-modal';
+import { Modal as PatternFlyModal, ModalVariant } from '@patternfly/react-core';
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ui/src/components/Modal.tsx:2]: Then I would need to rename `ui/src/components/Modal.tsx` to something right? And the best I can come up with is `InfraModal`.

File: ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx [hunk 3]
```diff
@@ -45,7 +43,9 @@ export default function DeleteClusterModal({ cluster, onCancel, onDeleted }: Pro
         header={`Are you sure you want to delete ${cluster.ID}?`}
         buttons={buttons}
       >
-        <span className="text-xl">This action cannot be undone.</span>
+        <span className="pf-u-warning-color-100 pf-u-font-size-2xl">
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx]: done.

File: ui/src/containers/ClusterInfoPage/MutableLifespan.tsx [hunk 4]
```diff
@@ -27,10 +26,7 @@ export default function MutableLifespan({ cluster }: Props): ReactElement {
         header="Cannot change the cluster lifespan"
         onAcknowledged={(): void => setError(null)}
       >
-        <div className="flex items-center">
-          <AlertCircle size={16} className="mr-2 text-alert-600" />
-          <span className="text-lg text-alert-600">{message}</span>
-        </div>
+        <span className="pf-u-warning-color-100 pf-u-font-size-2xl">{message}</span>
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ui/src/containers/ClusterInfoPage/MutableLifespan.tsx]: done.

File: ui/src/components/Modal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (vjwilson) [reply ui/src/components/Modal.tsx:2]: You don't have to make the local component's name unique, as long you don't import them both into the same file.

[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply ui/src/components/Modal.tsx]: ahh. done.

**Comment by gavin-stackrox** [comment]
@vjwilson how does this look now? I'm hoping to do a bit more clean out of tailwind today.

File: ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx [hunk 5]
```diff
@@ -66,10 +64,7 @@ export default function DeleteClusterModal({ cluster, onCancel, onDeleted }: Pro
     const message = `Cannot delete cluster. Server error occurred: "${error.message}".`;
     return (
       <InformationalModal header={`Failed to delete ${cluster.ID}!`} onAcknowledged={onCancel}>
-        <div className="flex items-center">
-          <AlertCircle size={16} className="mr-2 text-alert-600" />
-          <span className="text-lg text-alert-600">{message}</span>
-        </div>
+        <span className="pf-u-warning-color-100 pf-u-font-size-2xl">{message}</span>
```
Comment (vjwilson) [inline ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx:67]: My first reaction is that this color needs to be "danger" (red), but then I wonder if Failed to Delete is a blocker, or just annoying but not actionable by the end user.

Do you want to suggest they post this in the #team-automation channel, just to document zombies that may be lying around?

**Review by vjwilson — APPROVED** [review]

File: ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx (hunk 5 above)
Comment (gavin-stackrox) [reply ui/src/containers/ClusterInfoPage/DeleteClusterModal.tsx:67]: I was going to add the s/-warning-/-danger-/ but auto merge!

**Pull request merged**